Services/EntitiesServices/StudentServices/StudentService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using AutoMapper;
+using Domain.Entities;
+using Domain.EntitiesDto;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using Persistence.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services.EntitiesServices.StudentServices
+{
+    public class StudentService:IStudentService
+    {
+        private readonly DataContext _cont;
+        private readonly IMapper _mapper;
+        private readonly IHostingEnvironment _webHostEnvironment;
+        public StudentService(DataContext context, IMapper mapper, IHostingEnvironment webHostEnvironment)
+        {
+            _cont = context;
+            _mapper = mapper;
+            _webHostEnvironment = webHostEnvironment;
+        }
+
+        public async Task<List<Student>> GetStudents()
+        {
+            return await _cont.Students.ToListAsync();
+        }
+
+        public async Task<StudentDto> GetStudentById(int Id)
+        {
+            var student = await (from s in _cont.Students
+                                    where s.Id == Id
+                                    select new StudentDto
+                                    {
+                                        Id = s.Id,
+                                        FirstName = s.FirstName,
+                                        LastName = s.LastName,
+                                        ImageName = s.Image,
+                                        Description = s.Description,
+                                        Title = s.Title,
+                                        Address = s.Address,
+                                        Enabled = s.Enabled,
+                                        PhoneNumber = s.PhoneNumber,
+                                        Age = s.Age,
+
+                                    }).FirstOrDefaultAsync();
+            if (student == null) return new StudentDto();
+            return student;
+        }
+
+        public async Task<int> Insert(StudentDto student)
+        {
+            var fileName = Guid.NewGuid() + "_" + Path.GetFileName(student.Image.FileName);
+            var path = Path.Combine(_webHostEnvironment.WebRootPath, "Images", fileName);
+            using (var stream = new FileStream(path, FileMode.Create))
+            {
+                await student.Image.CopyToAsync(stream);
+            }
+            var mapped = _mapper.Map<Student>(student);
+            mapped.Image = fileName;
+            await _cont.Students.AddAsync(mapped);
+            return await _cont.SaveChangesAsync();
+        }
+        public async Task<int> Update(StudentDto student)
+        {
+            var fileName = Guid.NewGuid() + "_" + Path.GetFileName(student.Image.FileName);
```

Comment:
Check if image is not null 

if image is not null then delete the old image and add new one 
otherwise dont do anithing to image